lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.h

```diff
@@ -606,6 +607,15 @@ class ProxyShape
   AL_USDMAYA_PUBLIC
   void deserialiseTranslatorContext();
 
+  /// \brief  gets the maya node path for a prim, stores the mapping and returns it
+  AL_USDMAYA_PUBLIC
+  MString recordUsdPrimToMayaPath(const UsdPrim &usdPrim,
+                                  const MObject &mayaObject);
```

Comment:
Please add /param's for each of the arguments. Also please add docs for the error conditions (e.g. "this method never fails, but it may return "dave"; or this method can fail based on X, Y, and Z, in which case it returns an empty string)